File: challs/injection_conf.dir/injection_conf.py
#!/usr/bin/python3
import sys, os
import re
import random, string

## Usage :
# ./injection_conf.py <directory_path>
# ex: ./injection_conf.py /tmp/test
# create a config file (configuration_file_name) containing a variable which value is set to <directory_path>
# then create a project based on this configuration file

## check params
if len(sys.argv) != 2 or not sys.argv[1]:
    print("Please send me a directory path so I can launch my configuration script !")
    sys.exit(0)
directory = sys.argv[1]
regex_dir = r"(/tmp/|/srv/writable/)[a-zA-Z0-9_-]+"
regex_dir_compiled = re.compile(regex_dir)
if not directory or not regex_dir_compiled.search(directory):
    print("Error, directory param not valid.")
    sys.exit(0)
if os.path.isdir(directory):
    print("Directory already exists.")
    sys.exit(0)
## end check params

## write configuration file
configuration_file_name = "/srv/writable/"
configuration_file_name += ''.join(random.choice(string.ascii_letters + string.digits) for _ in range(30))
configuration_file_name += "_config.sh"
with open(configuration_file_name, "w") as fh:
    fh.write("DATE=`date`\n")
    fh.write("DIRECTORY=" + directory + "\n")
## end write configuration file

## create project based on the config file
command = ". " + configuration_file_name + " && "
command += 'mkdir "${DIRECTORY}" && '
command += 'echo ${DATE} > "${DIRECTORY}/folder_configured"'
os.system(command)
## end create project based on the config file

if os.path.isdir(directory):
    print("Directory configured.")
else:
    print("Failed to configure directory.")

File: challs/injection_conf.dir/injection_conf.pl
#!/usr/bin/perl -w
use warnings;
use strict;

## Usage :
# ./injection_conf.pl <directory_path>
# ex: ./injection_conf.pl /tmp/test
# create a config file (configurationFileName) containing a variable which value is set to <directory_path>
# then create a project based on this configuration file

## check params
if (@ARGV == 0 || !$ARGV[0]) {
	print "Please send me a directory path so I can launch my configuration script !\n";
	exit 0;
}
my $directory = $ARGV[0];
if ($directory !~ /(\/tmp\/|\/srv\/writable\/)[a-zA-Z0-9_-]+/){
    print "Error, directory param not valid.\n";
    exit 0;
}
if (-e $directory){
    print "Directory already exists.\n";
    exit 0;
}
## end check params

## write configuration file
my @chars = ("A".."Z", "a".."z", "0".."9");
my $configurationFileName = "/srv/writable/";
$configurationFileName .= $chars[rand @chars] for 1..30;
$configurationFileName .= "_config.sh";
open(my $fh, '>', $configurationFileName) or die "Could not open file '$configurationFileName' $!";
print $fh "DATE=`date`\n";
print $fh "DIRECTORY=$directory\n";
close $fh;
## end write configuration file

## create project based on the config file
my $command = ". $configurationFileName && ";
$command .= 'mkdir "${DIRECTORY}" && ';
$command .= 'echo ${DATE} > "${DIRECTORY}/folder_configured"';
system($command);
## end create project based on the config file

if (-e $directory){
    print "Directory configured.\n";
}else{
    print "Failed to configure directory.\n";
}

1;
